clustering/server/src/main/java/org/wildfly/clustering/server/singleton/PrimaryProxyService.java

```diff
@@ -71,7 +71,10 @@ public T getValue() {
                 // This would mean there are multiple primary nodes!
                 throw ClusteringServerLogger.ROOT_LOGGER.multiplePrimaryProvidersDetected(this.context.getServiceName().getCanonicalName(), result.stream().map(Map.Entry::getKey).collect(Collectors.toList()));
             }
-            return result.stream().findFirst().orElseThrow(() -> ClusteringServerLogger.ROOT_LOGGER.noResponseFromMaster(this.context.getServiceName().getCanonicalName())).getValue().orElse(null);
+            for (Map.Entry<Node, Optional<T>> nodeOptionalEntry : result) {
```

Comment:
Fixing this, but this code needs a revisit because it uses many lambdas and many streams, for what is essentially a single iteration and could probably be simplified considerably.